tachyon/zk/air/plonky3/challenger/duplex_challenger_unittest.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 The Plonky3 Authors
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.plonky3 and the LICENCE-APACHE.plonky3
+// file.
+
+#include "tachyon/zk/air/plonky3/challenger/duplex_challenger.h"
+
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "tachyon/base/bits.h"
+#include "tachyon/crypto/hashes/sponge/poseidon2/poseidon2.h"
+#include "tachyon/crypto/hashes/sponge/poseidon2/poseidon2_plonky3_external_matrix.h"
+#include "tachyon/math/finite_fields/baby_bear/poseidon2.h"
+#include "tachyon/math/finite_fields/test/finite_field_test.h"
+
+namespace tachyon::zk::air::plonky3 {
+
+using F = math::BabyBear;
+using Poseidon2 = crypto::Poseidon2Sponge<
+    crypto::Poseidon2ExternalMatrix<crypto::Poseidon2Plonky3ExternalMatrix<F>>>;
+
+namespace {
+
+class DuplexChallengerTest : public math::FiniteFieldTest<F> {
+ public:
+  constexpr static size_t W = 16;
+  constexpr static size_t R = 4;
```

Comment:
Do these have to be named `kW` and `kR`?

Reply:
Then i'll use the full name.